nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -167,27 +171,32 @@ private int insertWorkflowInstanceWithTransaction(final WorkflowInstance instanc
       @Override
       public Integer doInTransaction(TransactionStatus status) {
         KeyHolder keyHolder = new GeneratedKeyHolder();
-        jdbc.update(new PreparedStatementCreator() {
-          @Override
-          @SuppressFBWarnings(value = "OBL_UNSATISFIED_OBLIGATION_EXCEPTION_EDGE", justification = "findbugs does not trust jdbctemplate")
-          public PreparedStatement createPreparedStatement(Connection connection) throws SQLException {
-            PreparedStatement ps;
-            int p = 1;
-            ps = connection.prepareStatement(insertWorkflowInstanceSql(), new String[] { "id" });
-            ps.setString(p++, instance.type);
-            ps.setObject(p++, instance.rootWorkflowId);
-            ps.setObject(p++, instance.parentWorkflowId);
-            ps.setObject(p++, instance.parentActionId);
-            ps.setString(p++, instance.businessKey);
-            ps.setString(p++, instance.externalId);
-            ps.setString(p++, executorInfo.getExecutorGroup());
-            ps.setString(p++, instance.status.name());
-            ps.setString(p++, instance.state);
-            ps.setString(p++, abbreviate(instance.stateText, instanceStateTextLength));
-            ps.setTimestamp(p++, toTimestamp(instance.nextActivation));
-            return ps;
-          }
-        }, keyHolder);
+        try {
+          jdbc.update(new PreparedStatementCreator() {
+            @Override
+            @SuppressFBWarnings(value = "OBL_UNSATISFIED_OBLIGATION_EXCEPTION_EDGE", justification = "findbugs does not trust jdbctemplate")
+            public PreparedStatement createPreparedStatement(Connection connection) throws SQLException {
+              PreparedStatement ps;
+              int p = 1;
+              ps = connection.prepareStatement(insertWorkflowInstanceSql(), new String[] { "id" });
+              ps.setString(p++, instance.type);
+              ps.setObject(p++, instance.rootWorkflowId);
+              ps.setObject(p++, instance.parentWorkflowId);
+              ps.setObject(p++, instance.parentActionId);
+              ps.setString(p++, instance.businessKey);
+              ps.setString(p++, instance.externalId);
+              ps.setString(p++, executorInfo.getExecutorGroup());
+              ps.setString(p++, instance.status.name());
+              ps.setString(p++, instance.state);
+              ps.setString(p++, abbreviate(instance.stateText, instanceStateTextLength));
+              ps.setTimestamp(p++, toTimestamp(instance.nextActivation));
+              return ps;
+            }
+          }, keyHolder);
+        } catch (DuplicateKeyException e) {
+          logger.warn("Failed to insert workflow instance", e);
+          return -1;
+        }
         int id = keyHolder.getKey().intValue();
```

Comment:
If it gets DuplicateKeyException above, this will insert variables again? => there will be double set of variables?

Is the end result same as with insertWorkflowInstanceWithCte() ?

Would it be better to not modify variables in case of duplicate key exception?